File: src/services/instructions/loadstore/LD_W_REGX_MREGY_XX.java
package emulator.source.loadstore

import emulator.engine.CpuContext
 import Instruction from '../Instruction'import REGISTER_VALUE_NAME_MAPPER from '@/constants/registers'

export default class LD_W_REGX_MREGY_XX extends Instruction {
	public LD_W_REGX_MREGY_XX(memory, address, source, 
 destination) {
		super(memory, address, source, destination)
		super.setArgument32()
		super.setAssembler("ld.w " + this.sdest + ", [" + this.ssource + " + 0x%08x]")
	}

	
	exec ({ context, memory }) {
		context.getReg(this.destination)  = getMemContent(context, fix(context.getReg(this.source)  + this.argument) / 2, fix(context.getReg(this.source)  + this.argument))
		context.pc  += 6
	}
}

File: src/services/instructions/loadstore/LdWRegXMRegYXX.js
import Instruction from '../Instruction'
import { REGISTER_VALUE_NAME_MAPPER } from '@/constants/registers'

const ASSEMBLER_INSTRUCTION_EXPRESSION = (sdestination, ssource) => `ld.w ${sdestination}, [${ssource} + %s]`

export default class LdWRegXMRegYXX extends Instruction {
	constructor (memory, address, source, destination, symbolTable) {
		super(memory, address, source, destination, symbolTable)
		super.setArgument32(memory)
		super.setAssembler(ASSEMBLER_INSTRUCTION_EXPRESSION(this.sdestination, this.ssource), symbolTable)
	}

	exec ({ context, memory }) {
		context[REGISTER_VALUE_NAME_MAPPER[this.destination]]  =
		  Instruction.getMemContent(
				context,
				Instruction.fix(context[REGISTER_VALUE_NAME_MAPPER[this.source]]  + this.argument) / 2,
				Instruction.fix(context[REGISTER_VALUE_NAME_MAPPER[this.source]]  + this.argument),
				memory
			)

		context.pc  += 6
	}
}
